cmake/modules/AddSwiftUnittests.cmake

```diff
@@ -48,8 +48,10 @@ function(add_swift_unittest test_dirname)
       "${android_system_libs}")
     set_property(TARGET "${test_dirname}" APPEND PROPERTY LINK_LIBRARIES "log")
   elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
-    set_property(TARGET "${test_dirname}" APPEND PROPERTY LINK_LIBRARIES
-      "atomic")
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|AMD64")
+      set_property(TARGET "${test_dirname}" APPEND_STRING PROPERTY COMPILE_FLAGS
```

Comment:
I think that this is better written as:

```cmake
target_compile_options(${test_dirname} PRIVATE
  -march=core2)
```

Reply:
No strong opinion, but it's done with `set_property` in the rest of the file, so maybe it's worth keeping it consistent?

Reply:
The `target_*` form is more concise, avoids string manipulations, and should reduce memory overheads for CMake having to translate between representations.  It also will make it easier to split apart the host and target builds.

Reply:
Ok, changed it.